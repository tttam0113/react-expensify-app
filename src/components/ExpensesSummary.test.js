import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from './ExpensesSummary';

test('should correctly render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} />);
    expect(wrapper).toMatchSnapshot();
});


test('should correctly render ExpensesSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={23512312451} />);
    expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpensesSummary with 1 hidden expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} hiddenExpenseCount={1}/>);
    expect(wrapper).toMatchSnapshot();
})

test('should correctly render ExpensesSummary with multi hidden expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} hiddenExpenseCount={123}/>);
    expect(wrapper).toMatchSnapshot();
})
