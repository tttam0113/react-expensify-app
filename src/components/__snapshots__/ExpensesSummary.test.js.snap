// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling
       
      <span>
        $2.35
      </span>
    </h1>
    <h3
      className="page-header__subtitle"
    >
      Hidden 
      <span>
        0
      </span>
       
      expenses
    </h3>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should correctly render ExpensesSummary with 1 hidden expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling
       
      <span>
        $2.35
      </span>
    </h1>
    <h3
      className="page-header__subtitle"
    >
      Hidden 
      <span>
        1
      </span>
       
      expense
    </h3>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should correctly render ExpensesSummary with multi hidden expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling
       
      <span>
        $2.35
      </span>
    </h1>
    <h3
      className="page-header__subtitle"
    >
      Hidden 
      <span>
        123
      </span>
       
      expenses
    </h3>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should correctly render ExpensesSummary with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        23
      </span>
       
      expenses
       totalling
       
      <span>
        $235,123,124.51
      </span>
    </h1>
    <h3
      className="page-header__subtitle"
    >
      Hidden 
      <span>
        0
      </span>
       
      expenses
    </h3>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
